File: backend/app/schemas/equipment.py
from typing import Optional
from datetime import datetime
from uuid import UUID
from pydantic import BaseModel, Field


class ModuleBase(BaseModel):
    manufacturer: str = Field(..., min_length=1, max_length=255)
    model: str = Field(..., min_length=1, max_length=255)
    wattage: int = Field(..., gt=0)
    efficiency: float = Field(..., gt=0, le=100)
    
    length_m: float = Field(..., gt=0)
    width_m: float = Field(..., gt=0)
    thickness_m: float = Field(..., gt=0)
    
    voc: float = Field(..., gt=0)
    isc: float = Field(..., gt=0)
    vmp: float = Field(..., gt=0)
    imp: float = Field(..., gt=0)


class ModuleCreate(ModuleBase):
    pass


class ModuleUpdate(BaseModel):
    manufacturer: Optional[str] = Field(None, min_length=1, max_length=255)
    model: Optional[str] = Field(None, min_length=1, max_length=255)
    wattage: Optional[int] = Field(None, gt=0)
    efficiency: Optional[float] = Field(None, gt=0, le=100)
    length_m: Optional[float] = Field(None, gt=0)
    width_m: Optional[float] = Field(None, gt=0)
    thickness_m: Optional[float] = Field(None, gt=0)
    voc: Optional[float] = Field(None, gt=0)
    isc: Optional[float] = Field(None, gt=0)
    vmp: Optional[float] = Field(None, gt=0)
    imp: Optional[float] = Field(None, gt=0)
    is_active: Optional[bool] = None


class ModuleResponse(ModuleBase):
    id: UUID
    tenant_id: Optional[UUID]
    is_global: bool
    is_active: bool
    created_at: datetime
    
    class Config:
        from_attributes = True


class InverterBase(BaseModel):
    manufacturer: str = Field(..., min_length=1, max_length=255)
    model: str = Field(..., min_length=1, max_length=255)
    capacity_kw: float = Field(..., gt=0)
    
    max_dc_voltage: float = Field(..., gt=0)
    mppt_voltage_range_min: float = Field(..., gt=0)
    mppt_voltage_range_max: float = Field(..., gt=0)
    max_input_current: float = Field(..., gt=0)
    num_mppt_channels: int = Field(..., gt=0)


class InverterCreate(InverterBase):
    pass


class InverterUpdate(BaseModel):
    manufacturer: Optional[str] = Field(None, min_length=1, max_length=255)
    model: Optional[str] = Field(None, min_length=1, max_length=255)
    capacity_kw: Optional[float] = Field(None, gt=0)
    max_dc_voltage: Optional[float] = Field(None, gt=0)
    mppt_voltage_range_min: Optional[float] = Field(None, gt=0)
    mppt_voltage_range_max: Optional[float] = Field(None, gt=0)
    max_input_current: Optional[float] = Field(None, gt=0)
    num_mppt_channels: Optional[int] = Field(None, gt=0)
    is_active: Optional[bool] = None


class InverterResponse(InverterBase):
    id: UUID
    tenant_id: Optional[UUID]
    is_global: bool
    is_active: bool
    created_at: datetime
    
    class Config:
        from_attributes = True

File: frontend/src/test/fixtures/equipment.ts
import { EquipmentModule, EquipmentInverter } from '@/types';

export const mockEquipmentModule: EquipmentModule = {
    id: 'module-1',
    manufacturer: 'Test Solar',
    model: 'TS-400',
    wattage: 400,
    efficiency: 21.5,
    length_m: 2.0,
    width_m: 1.0,
    thickness_m: 0.04,
    voc: 45.5,
    isc: 11.2,
    vmp: 38.4,
    imp: 10.4,
    tenant_id: 'tenant-1',
    is_global: true,
    is_active: true,
    created_at: '2024-01-01T00:00:00Z',
};

export const mockEquipmentInverter: EquipmentInverter = {
    id: 'inverter-1',
    manufacturer: 'Test Energy',
    model: 'TE-50K',
    capacity_kw: 50,
    max_dc_voltage: 1000,
    mppt_voltage_range_min: 200,
    mppt_voltage_range_max: 800,
    max_input_current: 50,
    num_mppt_channels: 4,
    tenant_id: 'tenant-1',
    is_global: true,
    is_active: true,
    created_at: '2024-01-01T00:00:00Z',
};

export const mockModulesList = [
    mockEquipmentModule,
    {
        ...mockEquipmentModule,
        id: 'module-2',
        model: 'TS-450',
        wattage: 450
    }
];

export const mockInvertersList = [
    mockEquipmentInverter,
    {
        ...mockEquipmentInverter,
        id: 'inverter-2',
        model: 'TE-100K',
        capacity_kw: 100
    }
];
